From 888b91f1db429856a73e760b5d6c8edc43263e30 Mon Sep 17 00:00:00 2001
From: Kendall Willis <k-willis@ti.com>
Date: Mon, 1 Dec 2025 15:36:08 -0600
Subject: [PATCH 2/2] perf(ti): Decrease CPU speed for max power savings

Decrease the CPU clock speed to 416.67 MHz in order to maximize the
power saved.

Change-Id: I0beb51474a1f22ddd37c861dac9f9f6a8ca4d658
Signed-off-by: Kendall Willis <k-willis@ti.com>
---
 plat/ti/k3/common/k3_bl1_setup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/ti/k3/common/k3_bl1_setup.c b/plat/ti/k3/common/k3_bl1_setup.c
index 5dc812e7f..ea8ae0c9c 100644
--- a/plat/ti/k3/common/k3_bl1_setup.c
+++ b/plat/ti/k3/common/k3_bl1_setup.c
@@ -73,7 +73,7 @@ void bl1_early_platform_setup(void)
 	k3_console_setup();
 
 	/* PLL8_HSDIV0 feeds A53, bump it up to 1.25GHz, VCO @2.5GHz */
-	mmio_write_32(MAIN_PLL_MMR_BASE + MAIN_PLL_MMR_CFG_PLL8_HSDIV_CTRL0, 0x8001);
+	mmio_write_32(MAIN_PLL_MMR_BASE + MAIN_PLL_MMR_CFG_PLL8_HSDIV_CTRL0, 0x8005);
 }
 
 /******************************************************************************
-- 
2.34.1

